File: welpen/02-Backend/Puppies/src/main/java/de/irahi/welp/entity/Category.java
package de.irahi.welp.entity;

import java.util.HashSet;
import java.util.Objects;
import java.util.Set;

import jakarta.persistence.Access;
import jakarta.persistence.AccessType;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToMany;
import lombok.Data;

@Entity
@Data
@Access(AccessType.FIELD)
public class Category {
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	private Long id;
	
	@Column(name="category_name")
	private String categoryName;
	
	@OneToMany(mappedBy="category", cascade=CascadeType.ALL, fetch= FetchType.EAGER )
	private Set<ProductObject> productObject;
	
	@Override
	public int hashCode() {
		return Objects.hash(id, categoryName, productObject);
	}

}

File: 03-Frontend/welps/src/app/common/category.ts
export class Category {

    constructor(private _id: number, private _categoryName:string){

    }

    get id():number | undefined{ return this._id; }
    get categoryName(): string | undefined {  return this._categoryName;  }

    set id(id:number){  this._id = id; }
    set categoryName(category: string){ this._categoryName = category;}

    toString():string{  return `Id: ${this._id};\n Category-Name: ${this._categoryName}`;}

}
